README.md

```diff
@@ -1,18 +1,77 @@
 # ULCRS
-ULCRS, or Undergraduate Learning Center (ULC) Resource Scheduler, is a tutor scheduling desktop application intended for the ULC at UW-Madison. ULCRS is a project for the CS 506: Software Engineering course during Fall 2017.
+ULCRS, or Undergraduate Learning Center (ULC) Resource Scheduler, is a tutor scheduling desktop application intended for the ULC at UW-Madison. ULCRS is a project for the CS 506: Software Engineering course during the Fall 2017 semester.
+
+The intent of ULCRS is to schedule 20-40 tutors to 50+ courses for the ULC's Drop-in Tutoring service. Here, tutors are scheduled to courses, rather than a typical tutor-to-student scheduling. Students needing help can then "drop in" for help with a course during that course's scheduled time. ULCRS aims to addresses this scheduling problem for the ULC.
+
+About the name: Yes, it is pronounced "ulcers." Our client at the ULC has always wanted to have some software called "ULCers," which would incorporate their name of ULC in a comical way. We said "heck, we can do that," and ULCRS was born.
+
+
+## Overview
+ULCRS  mimics a webapp, but localized on a desktop. ULCRS has frontend and backend local processes that communicate using a RESTful API over localhost. ULCRS uses the [Electron](https://electron.atom.io/) framework to function as a cross-platform desktop application which uses technologies as a webapp does, but on the desktop. Accordingly, ULCRS uses React for the UI hosted within Electron. For the backend, ULCRS (i.e. Electron) spins off a separate Java process to act as a backend server. [Spark](http://sparkjava.com/) is used to handle the REST API and spin up the actual Jetty server for the backend. So, Electron acts as the UI/frontend and Spark is used to create a backend Java server. 
+
 
 ## Building
-To grab dependencies and run the UI, run:
+### Electron UI
+**Dependencies:** TODO. npm
+
+To grab dependencies and build the UI, from the top-level directory, run:
 ``` 
 cd electron-src
-npm install
-npm start
+sudo npm install
```

Comment:
Nitpick, but sudo is not necessary to install packages locally. If you do, we've got problems...

Reply:
I agree, it shouldn't be. I didn't say this, but the frontend section will still be modified after/during the frontend merge to master. I was having difficulties installing on Ubuntu when I was attempting to build fresh, but sudo helped (??), so it was left here.